### ISSUE TITLE
System.InvalidOperationException: Insecure connection not allowed on unity 2022

### ISSUE THREAD

**Issue by TheCaveOfWonders** [GlitchEnzo/NuGetForUnity#486]
In Unity 2022, we always get the following error when we try to use this tool.

```
Error while refreshing NuGet packages list: System.TypeInitializationException: The type initializer for 'NugetForUnity.NugetHelper' threw an exception. ---> System.InvalidOperationException: Insecure connection not allowed
  at (wrapper managed-to-native) UnityEngine.Networking.UnityWebRequest.BeginWebRequest(UnityEngine.Networking.UnityWebRequest)
  at UnityEngine.Networking.UnityWebRequest.SendWebRequest () [0x00001] in <70db141e85df422b83f6abf73ad8c14b>:0 
  at NugetForUnity.NugetHelper.DownloadImage (System.String url) [0x000a7] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetHelper.cs:1667 
  at NugetForUnity.NugetPackage.FromNuspec (NugetForUnity.NuspecFile nuspec) [0x00086] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetPackage.cs:167 
  at NugetForUnity.NugetPackage.FromNupkgFile (System.String nupkgFilepath) [0x00001] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetPackage.cs:193 
  at NugetForUnity.NugetHelper.UpdateInstalledPackages () [0x0005d] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetHelper.cs:976 
  at NugetForUnity.NugetHelper.Restore () [0x00001] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetHelper.cs:1536 
  at NugetForUnity.NugetHelper..cctor () [0x00264] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetHelper.cs:149 
   --- End of inner exception stack trace ---
  at NugetForUnity.NugetWindow.Refresh (System.Boolean forceFullRefresh) [0x0001c] in .\Library\PackageCache\com.github-glitchenzo.nugetforunity@620a6ce1e3\Editor\NugetWindow.cs:293 
0x00007ff7911ac19d (Unity) StackWalker::GetCurrentCallstack
0x00007ff7911b1149 (Unity) StackWalker::ShowCallstack
0x00007ff79217b841 (Unity) GetStacktrace
0x00007ff7928371d2 (Unity) DebugStringToFile
0x00007ff790149886 (Unity) DebugLogHandler_CUSTOM_Internal_Log
0x000002752b2bda2a (Mono JIT Code) (wrapper managed-to-native) UnityEngine.DebugLogHandler:Internal_Log (UnityEngine.LogType,UnityEngine.LogOption,string,UnityEngine.Object)
0x000002752b2bd8bb (Mono JIT Code) UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
0x00000276e137d9ec (Mono JIT Code) UnityEngine.Logger:LogFormat (UnityEngine.LogType,string,object[])
0x00000276e137d7fb (Mono JIT Code) UnityEngine.Debug:LogErrorFormat (string,object[])
0x00000276e137d3eb (Mono JIT Code) NugetForUnity.NugetWindow:Refresh (bool) (at ./Library/PackageCache/com.github-glitchenzo.nugetforunity@620a6ce1e3/Editor/NugetWindow.cs:330)
0x00000276e137cab3 (Mono JIT Code) NugetForUnity.NugetWindow:OnEnable () (at ./Library/PackageCache/com.github-glitchenzo.nugetforunity@620a6ce1e3/Editor/NugetWindow.cs:277)
0x000002752f89f158 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr)
0x00007ffb963e0394 (mono-2.0-bdwgc) mono_jit_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/mini/mini-runtime.c:3445)
0x00007ffb9631eb44 (mono-2.0-bdwgc) do_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/metadata/object.c:3066)
0x00007ffb9631ecdc (mono-2.0-bdwgc) mono_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/metadata/object.c:3113)
0x00007ff7910d4144 (Unity) scripting_method_invoke
0x00007ff7910b4aa4 (Unity) ScriptingInvocation::Invoke
0x00007ff7910b4b8e (Unity) ScriptingInvocation::InvokeChecked
0x00007ff7911457d6 (Unity) SerializableManagedRef::CallMethod
0x00007ff79109b7e1 (Unity) MonoBehaviour::AwakeFromLoad
0x00007ff7910a81f4 (Unity) ResetAndApplyDefaultReferencesOnNewMonoBehaviour
0x00007ff7910b2bf5 (Unity) Scripting::CreateScriptableObjectWithType
0x00007ff790182d0b (Unity) ScriptableObject_CUSTOM_CreateScriptableObjectInstanceFromType
0x000002752b14736c (Mono JIT Code) (wrapper managed-to-native) UnityEngine.ScriptableObject:CreateScriptableObjectInstanceFromType (System.Type,bool)
0x000002752b1471d3 (Mono JIT Code) UnityEngine.ScriptableObject:CreateInstance (System.Type)
0x00000276e137b903 (Mono JIT Code) UnityEditor.EditorWindow:GetWindowPrivate (System.Type,bool,string,bool)
0x00000276e137b5a3 (Mono JIT Code) UnityEditor.EditorWindow:GetWindow (System.Type,bool,string,bool)
0x00000276e137b3db (Mono JIT Code) UnityEditor.EditorWindow:GetWindow<T_REF> (bool,string,bool)
0x00000276e137b24b (Mono JIT Code) UnityEditor.EditorWindow:GetWindow<T_REF> ()
0x00000276e137b113 (Mono JIT Code) NugetForUnity.NugetWindow:DisplayNugetWindow () (at ./Library/PackageCache/com.github-glitchenzo.nugetforunity@620a6ce1e3/Editor/NugetWindow.cs:153)
0x000002752e3cf865 (Mono JIT Code) (wrapper runtime-invoke) object:runtime_invoke_void (object,intptr,intptr,intptr)
0x00007ffb963e0394 (mono-2.0-bdwgc) mono_jit_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/mini/mini-runtime.c:3445)
0x00007ffb9631eb44 (mono-2.0-bdwgc) do_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/metadata/object.c:3066)
0x00007ffb9631ecdc (mono-2.0-bdwgc) mono_runtime_invoke (at C:/build/output/Unity-Technologies/mono/mono/metadata/object.c:3113)
0x00007ff7910d4144 (Unity) scripting_method_invoke
0x00007ff7910b4aa4 (Unity) ScriptingInvocation::Invoke
0x00007ff7910b4b8e (Unity) ScriptingInvocation::InvokeChecked
0x00007ff791f1e8d1 (Unity) ScriptCommands::InvokeMenuItemWithContext
0x00007ff791f1bdf6 (Unity) ScriptCommands::Execute
0x00007ff791b12de2 (Unity) MenuController::ExecuteMenuItem
0x00007ff79218254f (Unity) ProcessMainWindowMessages
0x00007ff7921a40ce (Unity) ContainerWindow::ContainerWndProc
0x00007ffc534ae858 (USER32) CallWindowProcW
0x00007ffc534ae299 (USER32) DispatchMessageW
0x00007ff792181383 (Unity) MainMessageLoop
0x00007ff792185830 (Unity) WinMain
0x00007ff79354c9ae (Unity) __scrt_common_main_seh
0x00007ffc539e7034 (KERNEL32) BaseThreadInitThunk
0x00007ffc55022651 (ntdll) RtlUserThreadStart
```

**Comment by JoC0de** [comment]
Looks like a issue with http instead of https. So please try replacing http://www.nuget.org/api/v2/ with https://www.nuget.org/api/v2/ inside the nugget.config file in the line:
<add key="NuGet" value="http://www.nuget.org/api/v2/" />

**Comment by JoC0de** [comment]
CheckForUpdates() is only called when you click on check for updates in the UI. It could be that the code change forced the reload of the nuget.config file.

**Comment by Ero98** [comment]
I have the same issue that Unity Console complaining about connection not secure when installing some packages. 
I found out that these packages are actually downloaded into `Assets/Packages`  folder, and my code actually compiles fine. Only the `Installed` tab inside NuGet panel is missing these package info.

**Comment by JoC0de** [comment]
I think the error occurs on the download of the package Icon (when looking at the stack-trace). 
1. It would be interesting what package has this issue eventually they are using some special URLs.
2. I will implement a ignore / warn only if package icon download fails currently its a exception and so it doesn't continue with the normal installation.

**Comment by JoC0de** [comment]
Should be fixed by #493 / the errors should be ignored

**Comment by JoC0de** [comment]
It should now also be fixed in the latest release (includes .unitypackage) version 3.1.0. I only see errors when NuGetForUnity tries to download package icons, as the URL is defined by each package, which is now fault tolerant. The URL used to download the packages itself can be changed inside `NuGet.config`.
If nothing helps there is also a setting in Unity "Edit -> Project Settings -> Player -> Other Settings -> Configuration -> Allow downloads over HTTP"
